modules/javafx.graphics/.classpath

```diff
@@ -1,6 +1,8 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <classpath>
 	<classpathentry kind="src" path="src/main/java"/>
+	<classpathentry kind="src" path="build/gensrc/jsl-prism"/>
+	<classpathentry kind="src" path="build/gensrc/jsl-decora"/>
```

Comment:
these two seem not enough for running rojects that depend on graphics, without the other two (from the original before the [PR 804](https://github.com/openjdk/jfx/pull/804) I'm still getting runtime exceptions (though different from those copied shown on the [mailinglist](https://mail.openjdk.org/pipermail/openjfx-dev/2022-July/034806.html, see below).

It's only working with all four of the original, that is 

	<classpathentry kind="src" path="build/hlsl/Prism">
		<attributes>
			<attribute name="ignore_optional_problems" value="true"/>
		</attributes>
	</classpathentry>
	<classpathentry kind="src" path="build/gensrc/jsl-prism"/>
	<classpathentry kind="src" path="build/hlsl/Decora">
		<attributes>
			<attribute name="ignore_optional_problems" value="true"/>
		</attributes>
	</classpathentry>
	<classpathentry kind="src" path="build/gensrc/jsl-decora"/>
	
stacktrace if both build/hsls/xx are missing:

```
java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at javafx.graphics/com.sun.prism.d3d.D3DResourceFactory.createStockShader(D3DResourceFactory.java:429)
	at javafx.graphics/com.sun.prism.impl.ps.BaseShaderContext.getPaintShader(BaseShaderContext.java:269)
	at javafx.graphics/com.sun.prism.impl.ps.BaseShaderContext.validatePaintOp(BaseShaderContext.java:500)
	at javafx.graphics/com.sun.prism.impl.ps.BaseShaderContext.validatePaintOp(BaseShaderContext.java:369)
	at javafx.graphics/com.sun.prism.impl.ps.BaseShaderGraphics.renderGeneralRoundedPgram(BaseShaderGraphics.java:919)
	at javafx.graphics/com.sun.prism.impl.ps.BaseShaderGraphics.renderGeneralRoundedRect(BaseShaderGraphics.java:620)
	at javafx.graphics/com.sun.prism.impl.ps.BaseShaderGraphics.fillRect(BaseShaderGraphics.java:1526)
	at javafx.graphics/com.sun.javafx.sg.prism.NGRegion.renderBackgroundRectanglesDirectly(NGRegion.java:1112)
	at javafx.graphics/com.sun.javafx.sg.prism.NGRegion.renderBackgroundRectangle(NGRegion.java:852)
	at javafx.graphics/com.sun.javafx.sg.prism.NGRegion.renderAsRectangle(NGRegion.java:754)
	at javafx.graphics/com.sun.javafx.sg.prism.NGRegion.renderContent(NGRegion.java:575)
	at javafx.graphics/com.sun.javafx.sg.prism.NGNode.doRender(NGNode.java:2072)
	at javafx.graphics/com.sun.javafx.sg.prism.NGNode.render(NGNode.java:1964)
	at javafx.graphics/com.sun.javafx.tk.quantum.ViewPainter.doPaint(ViewPainter.java:480)
	at javafx.graphics/com.sun.javafx.tk.quantum.ViewPainter.paintImpl(ViewPainter.java:329)
	at javafx.graphics/com.sun.javafx.tk.quantum.PresentingPainter.run(PresentingPainter.java:92)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
	at javafx.graphics/com.sun.javafx.tk.RenderJob.run(RenderJob.java:58)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at javafx.graphics/com.sun.javafx.tk.quantum.QuantumRenderer$PipelineRunnable.run(QuantumRenderer.java:126)
	at java.base/java.lang.Thread.run(Thread.java:835)
Caused by: java.lang.RuntimeException: InputStream must be non-null
	at javafx.graphics/com.sun.prism.d3d.D3DResourceFactory.getBuffer(D3DResourceFactory.java:365)
	at javafx.graphics/com.sun.prism.d3d.D3DResourceFactory.createShader(D3DResourceFactory.java:409)
	at javafx.graphics/com.sun.prism.shader.FillPgram_Color_Loader.loadShader(FillPgram_Color_Loader.java:47)
	... 27 more
java.lang.InternalError: Error loading stock shader FillPgram_Color
	at javafx.graphics/com.sun.prism.d3d.D3DResourceFactory.createStockShader(D3DResourceFactory.java:432)
	at javafx.graphics/com.sun.prism.impl.ps.BaseShaderContext.getPaintShader(BaseShaderContext.java:269)
	at javafx.graphics/com.sun.prism.impl.ps.BaseShaderContext.validatePaintOp(BaseShaderContext.java:500)
	at javafx.graphics/com.sun.prism.impl.ps.BaseShaderContext.validatePaintOp(BaseShaderContext.java:369)
	at javafx.graphics/com.sun.prism.impl.ps.BaseShaderGraphics.renderGeneralRoundedPgram(BaseShaderGraphics.java:919)
	at javafx.graphics/com.sun.prism.impl.ps.BaseShaderGraphics.renderGeneralRoundedRect(BaseShaderGraphics.java:620)
	at javafx.graphics/com.sun.prism.impl.ps.BaseShaderGraphics.fillRect(BaseShaderGraphics.java:1526)
	at javafx.graphics/com.sun.javafx.sg.prism.NGRegion.renderBackgroundRectanglesDirectly(NGRegion.java:1112)
	at javafx.graphics/com.sun.javafx.sg.prism.NGRegion.renderBackgroundRectangle(NGRegion.java:852)
	at javafx.graphics/com.sun.javafx.sg.prism.NGRegion.renderAsRectangle(NGRegion.java:754)
	at javafx.graphics/com.sun.javafx.sg.prism.NGRegion.renderContent(NGRegion.java:575)
	at javafx.graphics/com.sun.javafx.sg.prism.NGNode.doRender(NGNode.java:2072)
	at javafx.graphics/com.sun.javafx.sg.prism.NGNode.render(NGNode.java:1964)
	at javafx.graphics/com.sun.javafx.tk.quantum.ViewPainter.doPaint(ViewPainter.java:480)
	at javafx.graphics/com.sun.javafx.tk.quantum.ViewPainter.paintImpl(ViewPainter.java:329)
	at javafx.graphics/com.sun.javafx.tk.quantum.PresentingPainter.run(PresentingPainter.java:92)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
	at javafx.graphics/com.sun.javafx.tk.RenderJob.run(RenderJob.java:58)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at javafx.graphics/com.sun.javafx.tk.quantum.QuantumRenderer$PipelineRunnable.run(QuantumRenderer.java:126)
	at java.base/java.lang.Thread.run(Thread.java:835)


```

Reply:
the tests are fine - it happens if I have a separate project and let that depend on the controls et al projects (just the same setup as described on the mailinglist)

Reply:
This is puzzling, @kleopatra .
I don't see a reference to [hlsl/]Prism or [hlsl/]Decora anywhere.  Not in any gradle.build file, not in every file.  I don't think these two directories are being built, though I can only test on Mac at the moment.

I have a separate (modular) project that builds and runs as well, and I don't see the exceptions (I am running against 8290473.apps branch).  I had to execute the 'sdk' target via gradle first, in order to generate stuff in build/gensrc/jsl-prism and build/gensrc/jsl-decora in graphics.

Reply:
hmm .. that's indeed weird - here (on win) both graphics/build/hlsl/Prism and graphics/build/hlsl/Decora are created by running ./gradlew sdk  ... will re-check

Reply:
did re-check (added your fork as remote, checkout this branch, run gradle clean and sdk): all four (as in original classpath) folders are under graphics build. And same errors as above when running a project that has the Eclipse fx projects base, graphics, controls on its module path.

Reply:
thank you for checking!

it looks like there is no way around creating these directories before the 'sdk' task in build.gradle, to silence the missing path warning in eclipse (and adding them back to the .classpath)

i was thinking of merely creating these two dirs in a new task 'initDirs' and make 'sdk' depend on that task, in the top level build.gradle.  Any objections, @kevinrushforth ?

Reply:
> i was thinking of merely creating these two dirs in a new task 'initDirs' and make 'sdk' depend on that task, in the top level build.gradle. Any objections

I suppose that would be OK. I presume you'll check that the presence of the empty dirs on Mac doesn't cause any problems?

Reply:
As mentioned in an earlier comment, the better way to address this is with an optional dependency on the generated sources.

Reply:
@kevinrushforth  : 
these empty directories created do not seem to pose a problem on Mac, all the tests are passing (incl. web ones).